apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -166,5 +167,12 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 
 	}
 
+	mode := strings.ToLower(string(r.Spec.Mode))
```

Comment:
This seems like an anti-pattern.

The constants are there so they can be directly used in comparison